types/client.go

```diff
@@ -246,13 +246,33 @@ type NodeListOptions struct {
 	Filter filters.Args
 }
 
+type ServiceCreateOptions struct {
```

Comment:
If this struct isn't going to have anything other than `EncodedRegistryAuth` in the future, should we name it something else? `ServiceCreateOptions` sounds like it should contain more options.